File: backend/ProjetoZ.Domain/Entities/Sorteio.cs
namespace ProjetoZ.Domain.Entities;

public class Sorteio
{
    public Guid Id { get; set; }

    public string Titulo { get; set; } = string.Empty;

    public string Descricao { get; set; } = string.Empty;

    // Prêmio: nível de VIP (opcional) e/ou produtos (opcional) — pelo menos um dos dois.
    public int? PremioVipNivel { get; set; }

    public List<Guid> PremioProdutoIds { get; set; } = new();

    public string Status { get; set; } = "aberto"; // aberto | sorteado

    public Guid? VencedorUserId { get; set; }

    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;

    public DateTime? SorteadoEm { get; set; }
}

File: frontend/app/models/Sorteio.ts
export interface SorteioProduto {
    id: string;
    nome: string;
    imagem: string;
}

export interface Sorteio {
    id: string;
    titulo: string;
    descricao: string;
    premioVipNivel: number | null;
    premioVipNivelNome: string | null;
    premioProdutos: SorteioProduto[];
    status: "aberto" | "sorteado";
    totalParticipantes: number;
    jaParticipando: boolean;
    vencedorNome: string | null;
    criadoEm: string;
    sorteadoEm: string | null;
}

export interface CreateSorteioRequest {
    titulo: string;
    descricao: string;
    premioVipNivel: number | null;
    premioProdutoIds: string[];
}
